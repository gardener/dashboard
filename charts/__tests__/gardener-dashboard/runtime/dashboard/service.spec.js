//
// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
//
// SPDX-License-Identifier: Apache-2.0
//

'use strict'

const { helm } = fixtures

const renderTemplates = helm.renderDashboardRuntimeTemplates

describe('gardener-dashboard', function () {
  describe('service', function () {
    let templates

    beforeEach(() => {
      templates = [
        'service'
      ]
    })

    it('should render the template', async function () {
      const values = {}
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [service] = documents
      expect(service).toMatchSnapshot()
    })

    it('should render the template with sessionAffinity None', async function () {
      const values = {
        global: {
          dashboard: {
            sessionAffinity: 'None'
          }
        }
      }
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [service] = documents
      expect(service.spec.sessionAffinity).toBe(values.global.dashboard.sessionAffinity)
    })
  })
})
