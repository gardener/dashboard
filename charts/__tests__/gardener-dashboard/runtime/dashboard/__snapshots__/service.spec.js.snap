// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gardener-dashboard service should render the template 1`] = `
{
  "apiVersion": "v1",
  "kind": "Service",
  "metadata": {
    "labels": {
      "app.kubernetes.io/component": "dashboard",
      "app.kubernetes.io/instance": "gardener-dashboard",
      "app.kubernetes.io/managed-by": "Helm",
      "app.kubernetes.io/name": "gardener-dashboard",
      "helm.sh/chart": "gardener-dashboard-runtime-0.1.0",
    },
    "name": "gardener-dashboard-service",
    "namespace": "garden",
  },
  "spec": {
    "ports": [
      {
        "name": "http",
        "port": 8080,
        "protocol": "TCP",
        "targetPort": 8080,
      },
    ],
    "selector": {
      "app.kubernetes.io/component": "dashboard",
      "app.kubernetes.io/name": "gardener-dashboard",
    },
    "sessionAffinity": "ClientIP",
    "type": "ClusterIP",
  },
}
`;
